dnf-behave-tests/features/steps/history.py

```diff
@@ -120,3 +122,26 @@ def step_impl(context):
                     '[history] package "{0}" matched as userinstalled.'.format(package))
         else:
             raise ValueError('Invalid action "{0}".'.format(action))
+
+
+@then('reasons are following')
+def step_impl(context):
+    check_context_table(context, ["Reason", "Package"])
+    cmd = " ".join(context.dnf.get_cmd(context) + ["repoquery", "--installed", "--qf='NEVRA:%{name}-%{epoch}:%{version}-%{release}.%{arch} REASON:%{reason}'"])
+    _, cmd_stdout, _ = run(cmd, shell=True, can_fail=True)
+
+    re_reason = re.compile(r"^NEVRA:(.*) REASON:(.*)$")
```

Comment:
I don't think the `repoquery --qf` command prints anything else than the formatted list. Unless it does, you could use a format like: `'%{name}-%{epoch}:%{version}-%{release}.%{arch} %{reason}'` and then leave out the whole regex and simply split by space to get the dict pair. Constructing the dict could then basically be a oneliner.